ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -361,6 +350,24 @@ static int component_select(struct ompi_win_t *win, void **base, size_t size, in
         worker_created = true;
     }
 
+    /* init request pool for request based operations */
+    if (mca_osc_ucx_component.requests_initialized == false) {
+        OBJ_CONSTRUCT(&mca_osc_ucx_component.requests, opal_free_list_t);
+        ret = opal_free_list_init (&mca_osc_ucx_component.requests,
```

Comment:
how does it make a difference? memory hooks are initialized during ucp_init()..

Reply:
Any reason we can't move the UCP context initialization from component open to here on first window create?  That's the intent of the change, but you're right, `ucp_init` is still being called in component open.

Reply:
Yossi is right, we need to move ucp_init to window creation. I am working on it now. Will push soon.